packages/gcc_tools.rb

```diff
@@ -17,12 +17,14 @@ def self.build
 
 # define constants
 CREW_PREFIX="${CREW_PREFIX:-/usr/local}"
-ARCH_ACTUAL="$(uname -m)"
-[[ "$ARCH_ACTUAL" == "armv8l" ]] && ARCH_ACTUAL="armv7l"
-ARCH="${ARCH:-$ARCH_ACTUAL}"
+ARCH="$(uname -m)"
 LIB_SUFFIX=""
 [ "${ARCH}" == "x86_64" ] && LIB_SUFFIX="64"
-CURL="LD_LIBRARY_PATH=/usr/lib${LIB_SUFFIX}:/lib${LIB_SUFFIX} /usr/bin/curl"
+if curl --version &>/dev/null; then
+  CURL=curl
+else
+  CURL="LD_LIBRARY_PATH=/usr/lib${LIB_SUFFIX}:/lib${LIB_SUFFIX} /usr/bin/curl"
+fi
```

Comment:
curl should always work so I wonder if this is just a Docker thing.

Reply:
There are definitely ways to break it. Try messing with glibc. (This is also a good way to deal with glibc breakage.)

Reply:
Speaking of which... this was useful when I broke glibc:
```
require 'package'

class Coreutils_static < Package
  description 'GNU coreutils built statically with musl'
  homepage 'http://www.gnu.org/software/coreutils/coreutils.html'
  version '8.32'
  compatibility 'all'
  source_url 'https://mirrors.sarata.com/gnu/coreutils/coreutils-8.32.tar.xz'
  source_sha256 '4458d8de7849df44ccab15e16b1548b285224dbba5f08fac070c1c0e0bcc4cfa'


  def self.build
  # static build script from https://github.com/luciusmagn/coreutils-static/blob/master/build.sh
system 'cat <<"EOF"> build.sh
#!/bin/bash
#
# build static coreutils because we need exercises in minimalism
# MIT licensed: google it or see robxu9.mit-license.org.
#
# For Linux, also builds musl for truly static linking.

coreutils_version="8.32"
musl_version="1.2.1"

platform=$(uname -s)

if [ -d build ]; then
  echo "= removing previous build directory"
  rm -rf build
fi

#mkdir build # make build directory
#pushd build

# download tarballs
#echo "= downloading coreutils"
#curl -LO http://ftp.gnu.org/gnu/coreutils/coreutils-${coreutils_version}.tar.xz

#echo "= extracting coreutils"
#tar xJf coreutils-${coreutils_version}.tar.xz

if [ "$platform" = "Linux" ]; then
  echo "= downloading musl"
  curl -LO http://www.musl-libc.org/releases/musl-${musl_version}.tar.gz

  echo "= extracting musl"
  tar -xf musl-${musl_version}.tar.gz

  echo "= building musl"
  working_dir=$(pwd)

  install_dir=${working_dir}/musl-install

  pushd musl-${musl_version}
  env CFLAGS="$CFLAGS -Os -ffunction-sections -fdata-sections -fPIC" LDFLAGS=\'-Wl,--gc-sections\' ./configure --prefix=${install_dir}
  make install
  popd # musl-${musl-version}

  echo "= setting CC to musl-gcc"
  export CC=${working_dir}/musl-install/bin/musl-gcc
  export CFLAGS="-static"
else
  echo "= WARNING: your platform does not support static binaries."
  echo "= (This is mainly due to non-static libc availability.)"
fi

echo "= building coreutils"

#pushd coreutils-${coreutils_version}
env FORCE_UNSAFE_CONFIGURE=1 CFLAGS="$CFLAGS -Os -ffunction-sections -fdata-sections -fPIC" LDFLAGS=\'-Wl,--gc-sections\' ./configure
make
#popd # coreutils-${coreutils_version}

popd # build

if [ ! -d releases ]; then
  mkdir releases
fi

echo "= striptease"
strip -s -R .comment -R .gnu.version --strip-unneeded build/coreutils-${coreutils_version}/coreutils
#echo "= compressing"

shopt -s extglob
# upx currently breaks the binaries
#for file in build/coreutils-${coreutils_version}/src/!(*.*)
#do
#	upx --ultra-brute $file
#done
echo "= extracting coreutils binary"
cp src/!(*.*) releases
echo "= done"
EOF'
    system 'chmod +x ./build.sh'
    system './build.sh'
    system 'curl -Lf https://salsa.debian.org/debian/debianutils/-/raw/master/which?inline=false > releases/which && chmod +x releases/which'
    system "sed -i 's#/bin/sh#/usr/local/bin/sh#g' releases/which"

  end

  def self.install
    system "mkdir -p #{CREW_DEST_PREFIX}/bin"
    FileUtils.cd("#{CREW_DEST_PREFIX}/bin") do
      system "echo '#!/bin/bash' > arch"
      system "echo '#{CREW_PREFIX}/bin/uname -m' >> arch"
      system 'chmod +x arch'
    end
    system "cp releases/* #{CREW_DEST_PREFIX}/bin/"
  end
end
```

Reply:
Also happy to just dump changes into a ```crewfix2``` as an alternate crewfix, or just create a separate package of utils...